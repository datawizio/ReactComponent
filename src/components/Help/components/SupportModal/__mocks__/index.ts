export const file = {
  "uid": "rc-upload-1624447080939-12",
  "lastModified": 1624346309471,
  "lastModifiedDate": "2021-06-22T07:18:29.471Z",
  "name": "main_logo.png",
  "size": 151625,
  "type": "image/png",
  "percent": 100,
  "originFileObj": {
    "uid": "rc-upload-1624447080939-12"
  },
  "status": "done",
  "response": {
    "upload": {
      "token": "Gw6zeacBaxwV6hmB2i5j1yIaD",
      "expires_at": "2021-06-26T12:32:08Z",
      "attachments": [
        {
          "url":
            "https://datawiz.zendesk.com/api/v2/attachments/371822025780.json",
          "id": 371822025780,
          "file_name": "main_logo.png",
          "content_url":
            "https://datawiz.zendesk.com/attachments/token/EV6As3zABAHRrPtIf7CrmQYHd/?name=main_logo.png",
          "mapped_content_url":
            "https://datawiz.zendesk.com/attachments/token/EV6As3zABAHRrPtIf7CrmQYHd/?name=main_logo.png",
          "content_type": "image/png",
          "size": 151625,
          "width": 669,
          "height": 670,
          "inline": false,
          "deleted": false,
          "thumbnails": [
            {
              "url":
                "https://datawiz.zendesk.com/api/v2/attachments/371822025900.json",
              "id": 371822025900,
              "file_name": "main_logo_thumb.png",
              "content_url":
                "https://datawiz.zendesk.com/attachments/token/IcekaptdzghWveJ401aETDZuh/?name=main_logo_thumb.png",
              "mapped_content_url":
                "https://datawiz.zendesk.com/attachments/token/IcekaptdzghWveJ401aETDZuh/?name=main_logo_thumb.png",
              "content_type": "image/png",
              "size": 2420,
              "width": 80,
              "height": 80,
              "inline": false,
              "deleted": false
            }
          ]
        }
      ],
      "attachment": {
        "url":
          "https://datawiz.zendesk.com/api/v2/attachments/371822025780.json",
        "id": 371822025780,
        "file_name": "main_logo.png",
        "content_url":
          "https://datawiz.zendesk.com/attachments/token/EV6As3zABAHRrPtIf7CrmQYHd/?name=main_logo.png",
        "mapped_content_url":
          "https://datawiz.zendesk.com/attachments/token/EV6As3zABAHRrPtIf7CrmQYHd/?name=main_logo.png",
        "content_type": "image/png",
        "size": 151625,
        "width": 669,
        "height": 670,
        "inline": false,
        "deleted": false,
        "thumbnails": [
          {
            "url":
              "https://datawiz.zendesk.com/api/v2/attachments/371822025900.json",
            "id": 371822025900,
            "file_name": "main_logo_thumb.png",
            "content_url":
              "https://datawiz.zendesk.com/attachments/token/IcekaptdzghWveJ401aETDZuh/?name=main_logo_thumb.png",
            "mapped_content_url":
              "https://datawiz.zendesk.com/attachments/token/IcekaptdzghWveJ401aETDZuh/?name=main_logo_thumb.png",
            "content_type": "image/png",
            "size": 2420,
            "width": 80,
            "height": 80,
            "inline": false,
            "deleted": false
          }
        ]
      }
    }
  },
  "xhr": {}
};
