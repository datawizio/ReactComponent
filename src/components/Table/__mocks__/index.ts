import { Action, BodyCellType, IColumn, TableState } from "../types";
import { withKnobs, text, boolean, number } from "@storybook/addon-knobs";
import { genColumns, genDataSource } from "../../../utils/data/dataGenerators";

const key = Math.random().toString(36).substring(7);

// defineCellType
export const validColumn: IColumn = {
  dtype: "dtype",
  dataIndex: "dataIndex",
  resizable: false
};

export const invalidColumn: IColumn = {
  dataIndex: "dataIndex",
  resizable: false
};

export const cell: BodyCellType = {
  dtype: "dtype",
  key: "key"
};

// filterByColumns
export const columnsMap: TableState["columnsMap"] = {
  col1: {
    dtype: "dtype",
    dataIndex: "1",
    resizable: true
  },
  col2: {
    dtype: "dtype2",
    dataIndex: "2",
    resizable: true
  },
  col3: {
    dtype: "dtype3",
    dataIndex: "3",
    resizable: true
  },
  col4: {
    dtype: "dtype4",
    dataIndex: "4",
    resizable: true
  }
};

export const obj: { [columnKey: string]: any } = {
  col1: {},
  col2: "any",
  col5: "string",
  col6: "number"
};

export const expectedOutput_filterByColumns = {
  col1: {},
  col2: "any"
};

// swapColumns
const generateCollumn = (isValid: boolean = true): IColumn => {
  const column = isValid ? { ...validColumn } : { ...invalidColumn };
  return {
    key,
    children: [
      {
        key,
        ...validColumn
      }
    ],
    ...column
  };
};

export const columnsMock: Array<IColumn> = [
  ...[...new Array(3)].map(() => generateCollumn(true)),
  ...[...new Array(2)].map(() => generateCollumn(false))
];

//Column

export const model = {
  key: "key",
  resizable: false,
  fixed: true,
  title: "title"
};

//Table

const columnsCount = number("columns count", 5);
const dataCount = number("data count", 5);
const treeColumns = boolean("treeColumns", true);
const treeData = boolean("treeData", true);
const toolBar = boolean("toolbar", true);
export const sortable = boolean("sortable", true);

export const columns = genColumns(columnsCount, treeColumns);

export let dataSource = genDataSource(dataCount, columns, ["string"], treeData);

if (columns[0]) {
  columns[0].fixed = "left";
  //
  columns[0].filters = [
    // @ts-ignore
    ...new Set(dataSource.map(item => item[columns[0].dataIndex]))
  ].map(item => ({
    text: item,
    value: item
  }));
}

// Handlers

export const dTypeConfig = {
  boolean: {},
  number: {},
  string: {}
};

// Table reducer

export const updateDataSourceAC = (payload): Action => ({
  type: "updateDataSource",
  payload: payload
});

export const updateColumnsAC = (payload): Action => ({
  type: "updateColumns",
  payload: payload || []
});

export const visibleColumnsKeysAC = (payload): Action => ({
  type: "visibleColumnsKeys",
  payload
});

export const paginateAC = (payload): Action => ({
  type: "paginate",
  payload
});

export const resetPaginationAC = (): Action => ({
  type: "resetPagination"
});

export const searchAC = (payload): Action => ({
  type: "search",
  payload
});

export const sorthAC = (payload): Action => ({
  type: "sort",
  payload: [].concat(payload)
});

export const filterAC = (payload): Action => ({
  type: "filter",
  payload
});

export const addLoadingRowAC = (payload): Action => ({
  type: "addLoadingRow",
  payload
});

export const setRowChildrenAC = (payload): Action => ({
  type: "setRowChildren",
  payload
});

export const swapColumnsAC = (payload): Action => ({
  type: "swapColumns",
  payload
});

export const expandRowAC = (payload): Action => ({
  type: "expandRow",
  payload
});

export const collapseRowAC = (payload): Action => ({
  type: "collapseRow",
  payload
});

export const loadingAC = (payload): Action => ({
  type: "loading",
  payload
});

export const recoveryStateAC = (payload): Action => ({
  type: "recoveryState",
  payload
});

export const updateRowAC = (payload): Action => ({
  type: "updateRow",
  payload
});

export const updateAC = (payload): Action => ({
  type: "update",
  payload
});
