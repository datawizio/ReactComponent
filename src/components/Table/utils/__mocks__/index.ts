import { BodyCellType, IColumn } from "../../types";

export const validColumn: IColumn = {
  dtype: "dtype",
  dataIndex: "dataIndex",
  resizable: false
};

export const cell: BodyCellType = {
  dtype: "dtype",
  key: "key"
};

export const invalidColumn: IColumn = {
  dataIndex: "dataIndex",
  resizable: false
};
