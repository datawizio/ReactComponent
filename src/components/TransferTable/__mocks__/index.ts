const mockTags = ["cat", "dog", "bird"];

export const mockData = [];
for (let i = 0; i < 20; i++) {
  mockData.push({
    key: i.toString(),
    title: `content${i + 1}`,
    description: `description of content${i + 1}`,
    disabled: i % 4 === 0,
    tag: mockTags[i % 3]
  });
}

export const originTargetKeys = mockData
  .filter(item => +item.key % 3 > 1)
  .map(item => item.key);

export const tableColumns = [
  {
    dataIndex: "title",
    title: "Name"
  },
  {
    dataIndex: "tag",
    title: "Tag",
    render: tag => tag
  },
  {
    dataIndex: "description",
    title: "Description",
    filters: [{ text: "test", value: "test" }]
  }
];
