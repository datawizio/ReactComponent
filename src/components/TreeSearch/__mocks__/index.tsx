export function generateData(x = 3, y = 2, z = 1, gData = []) {
  function _loop(_level, _preKey?, _tns?) {
    const preKey = _preKey || "0";
    const tns = _tns || gData;

    const children = [];
    for (let i = 0; i < x; i++) {
      const key = `${preKey}-${i}`;
      tns.push({ title: `${key}-label`, key: `${key}-key` });
      if (i < y) {
        children.push(key);
      }
    }
    if (_level < 0) {
      return tns;
    }
    const __level = _level - 1;
    children.forEach((key, index) => {
      tns[index].children = [];
      return _loop(__level, key, tns[index].children);
    });

    return null;
  }
  _loop(z);
  return gData;
}
