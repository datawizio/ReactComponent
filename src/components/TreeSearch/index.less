.tree-search-container {
  .tree-search-input {
    margin-bottom: 10px;
  }
}
