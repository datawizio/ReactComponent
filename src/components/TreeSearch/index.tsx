import React, {
  ReactText,
  useState,
  useCallback,
  useRef,
  useMemo,
  useEffect,
  useContext
} from "react";
import { Empty } from "antd";
import Tree, { TreeProps } from "../Tree";
import { AntTreeNodeProps } from "antd/lib/tree";
import SearchInput from "../SearchInput";
import { Key } from "rc-tree-select/es/interface";
import {
  flattenOptions,
  filterOptions
} from "rc-tree-select/es/utils/valueUtil";
import { unTree } from "../../utils/data/tree";
import "./index.less";
import ConfigContext from "../ConfigProvider/context";

export interface AntTreeNodePropsExtended extends AntTreeNodeProps {
  parentKey?: string;
}

export interface TreeSearchProps extends TreeProps {
  locale?: {
    /**
     * Placeholder для строки поиска
     */
    searchInputPlaceholder?: string;

    /**
     * Текст пустого результата при поиске
     */
    emptySearchResultText?: string;
  };

  /**
   * Строка поиска
   */
  searchValue?: string;

  /**
   * Функция, которая сетит строку поиска
   */
  setSearchValue?: (value: string) => void;

  /**
   * Показывать/не показывать строку поиска
   */
  showSearchInput?: boolean;

  /**
   * Функция сравнения, по умолчанию будет проверяться вхождение подстроки `value` в строку `node.title`
   */
  searchCondition?: (value: string, node: AntTreeNodeProps) => boolean;

  /**
   * Функция для передачи компонта поля Input
   */
  renderInput?: (handleSearchInputChange: (e: any) => void) => React.Component;
}

function defaultSearchCondition(searchValue, node) {
  return String(node.title)
    .toLowerCase()
    .includes(String(searchValue).toLowerCase());
}

const TreeSearch: React.FC<TreeSearchProps> = props => {
  const {
    showSearchInput,
    searchValue,
    setSearchValue,
    renderInput,
    defaultExpandedKeys,
    treeData,
    searchCondition,
    checkedKeys,
    onCheck,
    ...restProps
  } = props;

  const { searchInputPlaceholder, emptySearchResultText } = props.locale;

  const { translate } = useContext(ConfigContext);
  const timeout = useRef<NodeJS.Timeout>();

  const [internalCheckedKeys, setInternalCheckedKeys] = useState<ReactText[]>(
    //@ts-ignore
    checkedKeys
  );

  const [expandedKeys, setExpandedKeys] = useState<Key[]>(defaultExpandedKeys);
  const [allDisabled, setAllDisabled] = useState<boolean>(false);

  const flatData = useMemo(() => {
    return unTree(treeData);
  }, [treeData]);

  const internalSearchCondition = useMemo(() => {
    return searchCondition ? searchCondition : defaultSearchCondition;
  }, [searchCondition]);

  const mergedTreeData = filterOptions(searchValue, treeData, {
    optionFilterProp: "title",
    filterOption: internalSearchCondition
  });

  const handleSearchInputChange = useCallback(
    e => {
      const value = e.target.value;

      clearTimeout(timeout.current);

      timeout.current = setTimeout(() => {
        setSearchValue(value);
      }, 200);
    },
    [setSearchValue]
  );

  const handleTreeExpand = useCallback((keys: Key[]) => {
    setExpandedKeys(keys);
  }, []);

  const recDown = useCallback(
    (
      nodes: AntTreeNodePropsExtended[],
      value: boolean,
      nextCheckedKeysSet: Set<string | number>
    ) => {
      nodes.forEach(node => {
        if (node.disabled) return;

        value
          ? nextCheckedKeysSet.add(node.key)
          : nextCheckedKeysSet.delete(node.key);

        if (
          node.children &&
          Array.isArray(node.children) &&
          node.children.length
        ) {
          recDown(
            node.children as Array<AntTreeNodePropsExtended>,
            value,
            nextCheckedKeysSet
          );
        }
      });
    },
    []
  );

  const recUp = useCallback(
    (
      node: AntTreeNodePropsExtended,
      value: boolean,
      nextCheckedKeysSet: Set<string>
    ) => {
      const parent = flatData.find(
        item => item.key === node.parentKey || item.key === node.pId
      );
      if (!parent || !parent.key || parent.key === "-1") return;

      const checked: boolean = parent.children.every(child =>
        nextCheckedKeysSet.has(child.key)
      );

      checked
        ? nextCheckedKeysSet.add(parent.key)
        : nextCheckedKeysSet.delete(parent.key);

      recUp(parent, value, nextCheckedKeysSet);
    },
    [flatData]
  );

  const handleTreeCheck = useCallback(
    (keys, e) => {
      let nextKeys;
      let checked;

      const { node, event } = e;
      if (node.disabled) return;

      if (event === "select") {

        if (node.key === "-1" && node.checked) {
          setInternalCheckedKeys([]);
          onCheck && onCheck([], e);
          return;
        }

        const keyIdx = internalCheckedKeys.findIndex(item => item === node.key);
        checked = keyIdx === -1;

      } else {
        checked = e.checked;
      }

      if (Array.isArray(keys)) {
        nextKeys = [...keys];
      } else {
        nextKeys = [...keys.checked];
      }

      const nextCheckedKeysSet = new Set<string>(nextKeys);

      if (searchValue || event === "select") {
        checked
          ? nextCheckedKeysSet.add(node.key)
          : nextCheckedKeysSet.delete(node.key);

        if (node.children && node.children.length) {
          recDown(node.children, checked, nextCheckedKeysSet);
        }

        if (node.parentKey || node.pId) {
          recUp(node, checked, nextCheckedKeysSet);
        }

        nextKeys = Array.from(nextCheckedKeysSet);
      }

      const allIdx = nextKeys.findIndex(key => key === "-1");
      if (allIdx !== -1) nextKeys.splice(allIdx, 1);

      setInternalCheckedKeys(nextKeys);

      onCheck && onCheck(nextKeys, e);
    },
    [internalCheckedKeys, onCheck, recDown, recUp, searchValue]
  );

  const handleTreeSelect = useCallback((selectedKeys, e) => {
    handleTreeCheck(internalCheckedKeys, e);
  }, [handleTreeCheck, internalCheckedKeys]);

  useEffect(() => {
    const options = flattenOptions(treeData);
    if (searchValue) {
      setExpandedKeys(options.map(o => o.key));
      setAllDisabled(true);
    } else {
      setExpandedKeys(["-1"]);
      setAllDisabled(false);
    }
  }, [searchValue]);

  return (
    <div className="tree-search-container">
      <div className="tree-search-input" style={{ marginBottom: "15px" }}>
        {showSearchInput ? (
          renderInput ? (
            renderInput(handleSearchInputChange)
          ) : (
            <SearchInput
              onChange={handleSearchInputChange}
              placeholder={translate(searchInputPlaceholder)}
            />
          )
        ) : null}
      </div>
      {mergedTreeData.length ? (
        <Tree
          {...restProps}
          //@ts-ignore
          treeData={mergedTreeData}
          checkedKeys={internalCheckedKeys}
          expandedKeys={expandedKeys}
          selectedKeys={[]}
          checkStrictly={!!searchValue}
          isAllDisabled={allDisabled}
          onExpand={handleTreeExpand}
          onCheck={handleTreeCheck}
          onSelect={handleTreeSelect}
        />
      ) : (
        <Empty description={translate(emptySearchResultText)} />
      )}
    </div>
  );
};

TreeSearch.defaultProps = {
  locale: {
    searchInputPlaceholder: "SEARCH",
    emptySearchResultText: "NO_DATA"
  },
  showSearchInput: true,
  searchCondition: null,
  searchValue: "",
  setSearchValue: () => {}
};

export default TreeSearch;
