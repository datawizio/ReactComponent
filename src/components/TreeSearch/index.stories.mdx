import { Meta, Story, Preview, Props } from "@storybook/addon-docs/blocks";
import TreeSearch from "./index";
import Input from "../Input";
import { generateTreeList } from "../../utils/data/tree";
let treeData = generateTreeList();

<Meta title="Components/Tree/TreeSearch" component={TreeSearch} />

# TreeSearch

Компонент `TreeSearch` cлужит для вывода дерева `Tree` c поиском `Search`. Можно
выводить строку поиска или нет.

Внутри вшит алогритм фильтрации дерева.

## Basic

<Preview withToolbar>
  <Story name="Basic">
    <TreeSearch searchValue="1-" treeData={treeData} height={300} />
  </Story>
</Preview>

## With search input

<Preview withToolbar>
  <Story name="With search input">
    <TreeSearch showSearchInput treeData={treeData} height={300} />
  </Story>
</Preview>

## With search condition

<Preview withToolbar>
  <Story name="With search condition">
    <TreeSearch
      showSearchInput
      treeData={treeData}
      height={300}
      searchCondition={(value, node) =>
        node.title && node.title.endsWith(value)
      }
    />
  </Story>
</Preview>

## With custom input

<Preview withToolbar>
  <Story name="With custom input">
    <TreeSearch
      showSearchInput
      treeData={treeData}
      height={300}
      checkable
      renderInput={handleSearchInputChange => (
        <Input onChange={handleSearchInputChange} prefix="Search:" />
      )}
    />
  </Story>
</Preview>

## Props

<Props of={TreeSearch} />
