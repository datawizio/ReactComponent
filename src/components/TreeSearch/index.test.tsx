import "jsdom-global/register";
import React from "react";
import { mount, shallow } from "enzyme";
import { act } from "react-dom/test-utils";
import { generateData } from "./__mocks__";
import TreeSearch from "./index";
import { render, fireEvent, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";

const gData = generateData();

const mockProps = {
  treeData: gData,
  setSearchValue: jest.fn(),
  showCheckAll: true
};

const setUp = (props?) => shallow(<TreeSearch {...props} />);

describe("TreeSearch component", () => {
  let component;
  afterEach(cleanup);
  beforeEach(() => {
    component = setUp(mockProps);
  });

  it("should render correctly", async () => {
    expect(component).toMatchSnapshot();
    expect(component.find(".tree-search-input").length).toBeTruthy();
    expect(component.find("Tree").length).toBeTruthy();
  });

  it("should render correctly with empty data and without search", () => {
    const wrapper = setUp({
      ...mockProps,
      treeData: [],
      showSearchInput: false
    });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(".ant-input-search").length).toBeFalsy();
    expect(wrapper.find("Empty").length).toBeTruthy();
  });

  it("should render correctly with default expand keys", () => {
    const wrapper = setUp({
      ...mockProps,
      searchValue: "label",
      defaultExpandedKeys: ["0-0-0-key", "0-0-0-0-key", "0-0-key"]
    });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("Tree").prop("expandedKeys")).toStrictEqual([
      "0-0-0-key",
      "0-0-0-0-key",
      "0-0-key"
    ]);
  });

  it("should search changed", () => {
    const props = {
      ...mockProps,
      treeData: []
    };
    const wrapper = render(<TreeSearch {...props} />);
    const searchInput: any = wrapper.getByPlaceholderText("SEARCH");
    fireEvent.change(searchInput, {
      target: { value: "test" }
    });
    expect(searchInput.value).toBe("test");
  });

  it("should expand", () => {
    act(() => {
      component.find("Tree").first().props().onExpand(["0-0-key", "0-1-key"]);
    });
    component.update();
    expect(component.find("Tree").first().prop("expandedKeys")).toStrictEqual([
      "0-0-key",
      "0-1-key"
    ]);
  });

  it.skip("should check all items", () => {
    const props = {
      ...mockProps,
      treeData: [{ "key": "0-0-0-0-key", "title": "0-0-0-0-label" }],
      checkable: true
    };
    const wrapper = mount(<TreeSearch {...props} />);
    wrapper.find(".ant-tree-checkbox").first().simulate("click");
    expect(
      wrapper
        .find(".ant-tree-checkbox")
        .first()
        .hasClass("ant-tree-checkbox-checked")
    ).toBeTruthy();
  });
});
