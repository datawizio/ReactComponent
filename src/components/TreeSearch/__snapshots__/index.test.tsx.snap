// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TreeSearch component should render correctly 1`] = `
<div
  className="tree-search-container"
>
  <div
    className="tree-search-input"
    style={
      Object {
        "marginBottom": "15px",
      }
    }
  >
    <SearchInput
      onChange={[Function]}
      placeholder="SEARCH"
    />
  </div>
  <Tree
    checkAllKey="-1"
    checkAllTitle="Check All"
    checkStrictly={false}
    defaultExpandedKeys={Array []}
    expandedKeys={Array []}
    isAllDisabled={false}
    locale={
      Object {
        "emptySearchResultText": "NO_DATA",
        "searchInputPlaceholder": "SEARCH",
      }
    }
    onCheck={[Function]}
    onExpand={[Function]}
    onSelect={[Function]}
    selectedKeys={Array []}
    showCheckAll={true}
    treeData={
      Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "children": Array [],
                  "key": "0-0-0-0-key",
                  "title": "0-0-0-0-label",
                },
                Object {
                  "children": Array [],
                  "key": "0-0-0-1-key",
                  "title": "0-0-0-1-label",
                },
                Object {
                  "children": Array [],
                  "key": "0-0-0-2-key",
                  "title": "0-0-0-2-label",
                },
              ],
              "disabled": false,
              "key": "0-0-0-key",
              "title": "0-0-0-label",
            },
            Object {
              "children": Array [
                Object {
                  "children": Array [],
                  "key": "0-0-1-0-key",
                  "title": "0-0-1-0-label",
                },
                Object {
                  "children": Array [],
                  "key": "0-0-1-1-key",
                  "title": "0-0-1-1-label",
                },
                Object {
                  "children": Array [],
                  "key": "0-0-1-2-key",
                  "title": "0-0-1-2-label",
                },
              ],
              "disabled": false,
              "key": "0-0-1-key",
              "title": "0-0-1-label",
            },
            Object {
              "children": Array [],
              "key": "0-0-2-key",
              "title": "0-0-2-label",
            },
          ],
          "disabled": false,
          "key": "0-0-key",
          "title": "0-0-label",
        },
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "children": Array [],
                  "key": "0-1-0-0-key",
                  "title": "0-1-0-0-label",
                },
                Object {
                  "children": Array [],
                  "key": "0-1-0-1-key",
                  "title": "0-1-0-1-label",
                },
                Object {
                  "children": Array [],
                  "key": "0-1-0-2-key",
                  "title": "0-1-0-2-label",
                },
              ],
              "disabled": false,
              "key": "0-1-0-key",
              "title": "0-1-0-label",
            },
            Object {
              "children": Array [
                Object {
                  "children": Array [],
                  "key": "0-1-1-0-key",
                  "title": "0-1-1-0-label",
                },
                Object {
                  "children": Array [],
                  "key": "0-1-1-1-key",
                  "title": "0-1-1-1-label",
                },
                Object {
                  "children": Array [],
                  "key": "0-1-1-2-key",
                  "title": "0-1-1-2-label",
                },
              ],
              "disabled": false,
              "key": "0-1-1-key",
              "title": "0-1-1-label",
            },
            Object {
              "children": Array [],
              "key": "0-1-2-key",
              "title": "0-1-2-label",
            },
          ],
          "disabled": false,
          "key": "0-1-key",
          "title": "0-1-label",
        },
        Object {
          "children": Array [],
          "key": "0-2-key",
          "title": "0-2-label",
        },
      ]
    }
  />
</div>
`;

exports[`TreeSearch component should render correctly with default expand keys 1`] = `
<div
  className="tree-search-container"
>
  <div
    className="tree-search-input"
    style={
      Object {
        "marginBottom": "15px",
      }
    }
  >
    <SearchInput
      onChange={[Function]}
      placeholder="SEARCH"
    />
  </div>
  <Tree
    checkAllKey="-1"
    checkAllTitle="Check All"
    checkStrictly={true}
    defaultExpandedKeys={Array []}
    expandedKeys={
      Array [
        "0-0-0-key",
        "0-0-0-0-key",
        "0-0-key",
      ]
    }
    isAllDisabled={false}
    locale={
      Object {
        "emptySearchResultText": "NO_DATA",
        "searchInputPlaceholder": "SEARCH",
      }
    }
    onCheck={[Function]}
    onExpand={[Function]}
    onSelect={[Function]}
    selectedKeys={Array []}
    showCheckAll={true}
    treeData={
      Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "children": Array [],
                  "key": "0-0-0-0-key",
                  "title": "0-0-0-0-label",
                },
                Object {
                  "children": Array [],
                  "key": "0-0-0-1-key",
                  "title": "0-0-0-1-label",
                },
                Object {
                  "children": Array [],
                  "key": "0-0-0-2-key",
                  "title": "0-0-0-2-label",
                },
              ],
              "disabled": true,
              "key": "0-0-0-key",
              "title": "0-0-0-label",
            },
            Object {
              "children": Array [
                Object {
                  "children": Array [],
                  "key": "0-0-1-0-key",
                  "title": "0-0-1-0-label",
                },
                Object {
                  "children": Array [],
                  "key": "0-0-1-1-key",
                  "title": "0-0-1-1-label",
                },
                Object {
                  "children": Array [],
                  "key": "0-0-1-2-key",
                  "title": "0-0-1-2-label",
                },
              ],
              "disabled": true,
              "key": "0-0-1-key",
              "title": "0-0-1-label",
            },
            Object {
              "children": Array [],
              "key": "0-0-2-key",
              "title": "0-0-2-label",
            },
          ],
          "disabled": true,
          "key": "0-0-key",
          "title": "0-0-label",
        },
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "children": Array [],
                  "key": "0-1-0-0-key",
                  "title": "0-1-0-0-label",
                },
                Object {
                  "children": Array [],
                  "key": "0-1-0-1-key",
                  "title": "0-1-0-1-label",
                },
                Object {
                  "children": Array [],
                  "key": "0-1-0-2-key",
                  "title": "0-1-0-2-label",
                },
              ],
              "disabled": true,
              "key": "0-1-0-key",
              "title": "0-1-0-label",
            },
            Object {
              "children": Array [
                Object {
                  "children": Array [],
                  "key": "0-1-1-0-key",
                  "title": "0-1-1-0-label",
                },
                Object {
                  "children": Array [],
                  "key": "0-1-1-1-key",
                  "title": "0-1-1-1-label",
                },
                Object {
                  "children": Array [],
                  "key": "0-1-1-2-key",
                  "title": "0-1-1-2-label",
                },
              ],
              "disabled": true,
              "key": "0-1-1-key",
              "title": "0-1-1-label",
            },
            Object {
              "children": Array [],
              "key": "0-1-2-key",
              "title": "0-1-2-label",
            },
          ],
          "disabled": true,
          "key": "0-1-key",
          "title": "0-1-label",
        },
        Object {
          "children": Array [],
          "key": "0-2-key",
          "title": "0-2-label",
        },
      ]
    }
  />
</div>
`;

exports[`TreeSearch component should render correctly with empty data and without search 1`] = `
<div
  className="tree-search-container"
>
  <div
    className="tree-search-input"
    style={
      Object {
        "marginBottom": "15px",
      }
    }
  />
  <Empty
    description="NO_DATA"
  />
</div>
`;
